### PR TITLE
Fixed issue were int32 png files would cause system exceptions, just …

### PR DESCRIPTION
**Pull request by vanossj**
…convert everything to RGBA

### CODE REVIEW

**Comment by vanossj** [comment]
fixes #301.

Incorrect extension was a red herring, the actual bug was that the image file used int32 pixels, and [Pillow](https://github.com/python-pillow/Pillow/blob/b1c816169e1143f3c3752427e1b2712cce02d709/_imaging.c#L557) does not like setting the fill colour as a tuple for int32 pixels. Setting it as an integer seems to always be black.

Immediately converting the image to RGBA sidesteps having to support different pixel types.

File: memegen/domain/image.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def make_meme(top, bottom, font, background, path, match_font_size=False):
                         bottom, bottom_font, bottom_font_size)
 
     log.info("Generating image: %s", path)
-    return img.save(path, format=img.format)
+    return img.save(path, format='png')
```
Comment (jacebrowning) [inline memegen/domain/image.py]: What does this do to the generated images? Are they now PNGs pretending to be JPEGs?

Comment (jacebrowning) [inline memegen/domain/image.py]: It looks like the images are now several times bigger.

Comment (vanossj) [inline memegen/domain/image.py]: yes... what if we check before we force anything, and try to keep format if possible

File: memegen/domain/image.py [hunk 2]
```diff
@@ -48,6 +48,13 @@ def make_meme(top, bottom, font, background, path, match_font_size=False):
     """Add text to an image and save it."""
     log.info("Loading background: %s", background)
     img = ImageFile.open(background)
+    if img.mode not in ('RGB', 'RGBA'):
+        if img.format == 'JPEG':
+            img = img.convert('RGB')
```
Comment (jacebrowning) [inline memegen/domain/image.py]: I'm surprised to see that these lines are not hit by any tests (see the output of `$ make test` or https://travis-ci.org/jacebrowning/memegen/builds/181765988).

Is `'JPEG'` the correct label? Should it be `img.format in ('JPEG', 'JPG')`? Does case matter?

**Comment by vanossj** [comment]
[Pillow uses "JPEG" all caps](https://github.com/python-pillow/Pillow/blob/837d583b157cf92821059296e557f3d46497b6ff/PIL/JpegImagePlugin.py#L284), not JPG

added testing for added lines

**Comment by jacebrowning** [comment]
Thanks!

**Pull request merged**